src/go/k8s/main.go

```diff
@@ -98,6 +105,23 @@ func main() {
 
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	if debug {
+		go func() {
+			pprofMux := http.NewServeMux()
+			pprofMux.HandleFunc("/debug/pprof/", pprof.Index)
+			pprofMux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+			pprofMux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+			pprofMux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+			pprofMux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+			pprofServer := &http.Server{
+				Addr:              pprofAddr,
+				Handler:           pprofMux,
+				ReadHeaderTimeout: 3 * time.Second,
+			}
+			log.Fatal(pprofServer.ListenAndServe())
```

Comment:
quick question, would you want to also always bind it to `127.0.0.1` in case something like this gets deployed in a cloud so the service doesn't get owned. 

so that way for an admin on debug they'd have to `netcat` or some other tunnel 

forcing 127.0.0.1 as default with flag to override gives us a "safety" of accidents.

Reply:
If you don't enable the debug flag, this will not be served. If you do enable it, you're likely to want to add a service and scrape it.

It won't be exposed to the internet unless you create a type of service or ingress to do so. If you're doing that and we set it to localhost, you're overriding that anyway.

I'm not against it, but if you're using the default settings, it's not a security risk. If you enable it, it's still not a security risk without taking additional actions.

This is a common practice.

Reply:
Lgtm. Non blocking.